Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.mm

```diff
@@ -1102,6 +1102,21 @@ + (NSData *)getClientDataJSONForAuthenticationType:(_WKWebAuthenticationType)typ
     return clientDataJSON.autorelease();
 }
 
++ (NSData *)getClientDataJSONWithTopOrigin:(_WKWebAuthenticationType)type challenge:(NSData *)challenge origin:(NSString *)origin topOrigin:(NSString *)topOrigin crossOrigin:(BOOL)crossOrigin
+{
+    RetainPtr<NSData> clientDataJSON;
+
+#if ENABLE(WEB_AUTHN)
+    if (crossOrigin)
+        clientDataJSON = produceClientDataJson(type, challenge, origin, topOrigin, WebAuthn::Scope::CrossOrigin);
+    else
+        clientDataJSON = produceClientDataJson(type, challenge, origin, topOrigin);
```

Comment:
We could do all this on one line:
`clientDataJSON = produceClientDataJson(type, challenge, origin, topOrigin, crossOrigin ? WebAuthn::Scope::CrossOrigin: WebAuthn::Scope::SameOrigin);`